### PR TITLE
crush: cleanup update_device_class() log messages

### PR DESCRIPTION
**Pull request by guzhongyan**
Signed-off-by: Gu Zhongyan <guzhongyan@360.cn>

### CODE REVIEW

**Comment by guzhongyan** [comment]
$ceph osd crush set-device-class ssd all
osd.0 already set to class ssdosd.1 already set to class ssdosd.2 already set to class ssd

need ". " to separate each osd's output.

**Comment by tchaikov** [comment]
retest this please

File: src/crush/CrushWrapper.cc [hunk 1]
```diff
@@ -1905,7 +1905,7 @@ int CrushWrapper::update_device_class(int id,
   }
 
   if (class_map.count(id) != 0 && class_map[id] == class_id) {
-    *ss << name << " already set to class " << class_name;
+    *ss << name << " already set to class " << class_name << ". ";
```
Comment (tchaikov) [inline src/crush/CrushWrapper.cc]: i'd suggest add the separator in `OSDMonitor::prepare_command_impl()` instead. as there is chance that we put more output after this message,  if the osd's is already set to expected device class.

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -7510,6 +7510,10 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
                 << "' device_class '" << device_class << "'"
                 << dendl;
         err = newcrush.update_device_class(osd, device_class, name, &ss);
+        if (err == 0) {
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: @guzhongyan after a second thought, i think we should add the ". " in `update_device_class()`,  as it is difficult to figure out if that method has appended any error message to `ss` or not by checking its return value. and what i was looking at https://github.com/ceph/ceph/pull/21174/files#diff-0a5db46a44ae9900e226289a810f10e8R7524, where we need to add `". "` also.

sorry for the bikeshedding. thought it was simpler, but we can hardly cover all cases in that way.

Comment (guzhongyan) [reply src/mon/OSDMonitor.cc]: @tchaikov , updated, please take a look.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-04-12_15:56:07-rados-wip-kefu-testing-2018-04-12-2211-distro-basic-mira/

failed tests are 

- tracked by http://tracker.ceph.com/issues/23699
- caused by #19117, #21188
- fixed by #21406

**Pull request merged**